bundles/org.openhab.binding.mpd/src/main/java/org/openhab/binding/mpd/internal/protocol/MPDConnectionThread.java

```diff
@@ -0,0 +1,296 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.mpd.internal.protocol;
+
+import java.io.BufferedReader;
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.Socket;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.openhab.binding.mpd.internal.MPDException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Class for communicating with the music player daemon through a IP connection
+ *
+ * @author Stefan Röllin - Initial contribution
+ */
+@NonNullByDefault
+public class MPDConnectionThread extends Thread {
+
+    private static final int RECONNECTION_TIMEOUT = 60000; // 60 seconds
+
+    private final Logger logger = LoggerFactory.getLogger(MPDConnectionThread.class);
+
+    private final MPDResponseListener listener;
+
+    private final String address;
+    private final Integer port;
+    private final String password;
+
+    private @Nullable Socket socket = null;
+    private @Nullable InputStreamReader inputStreamReader = null;
+    private @Nullable BufferedReader reader = null;
+    private @Nullable DataOutputStream writer = null;
+
+    private final List<MPDCommand> pendingCommands = new ArrayList<>();
+    private AtomicBoolean isInIdle = new AtomicBoolean(false);
+    private AtomicBoolean disposed = new AtomicBoolean(false);
+
+    public MPDConnectionThread(MPDResponseListener listener, String address, Integer port, String password) {
```

Comment:
Named threads make debugging easier.
```suggestion
    public MPDConnectionThread(MPDResponseListener listener, String address, Integer port, String password) {
        super(MPDConnectionThread.class.getSimpleName() + " " + address + ":" + port);
```